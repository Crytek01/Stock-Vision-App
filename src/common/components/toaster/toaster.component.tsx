import { useToaster } from "@/common/contexts/toaster/hook/use-toaster.hook";
import { Snackbar, Alert, SxProps } from "@mui/material";
import { FunctionComponent } from "react";

interface IToaster {}

const ALERT_SX: SxProps = {
  width: "100%",
  padding: "24px",
  fontSize: "1.2rem",
};

export const Toaster: FunctionComponent<IToaster> = () => {
  const { messages } = useToaster();

  return (
    <Snackbar open={true}>
      <div>
        {messages.map(({ id, severity, message }) => (
          <Alert key={id} severity={severity} sx={ALERT_SX}>
            {message}
          </Alert>
        ))}
      </div>
    </Snackbar>
  );
};
