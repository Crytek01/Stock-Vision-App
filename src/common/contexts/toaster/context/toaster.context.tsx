import { createContext } from "react";

interface IToasterContext {
  messages: IInternalToasterMessage[];
  addMessage: (newMessage: IToasterMessage) => void;
}

export const ToasterContext = createContext<IToasterContext | null>(null);
