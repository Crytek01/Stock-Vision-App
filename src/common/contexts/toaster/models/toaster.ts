interface IToasterMessage {
  severity: "success" | "info" | "warning" | "error";
  message: string;
  durationMs?: number;
}

interface IInternalToasterMessage extends IToasterMessage {
  id: string;
}
