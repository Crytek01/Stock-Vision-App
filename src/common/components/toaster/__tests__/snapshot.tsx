import { render } from "@testing-library/react";
import { Toaster } from "../toaster.component";
import { ToasterProvider } from "@/common/contexts/toaster/provider/toaster.context";
it("renders tab panel component", () => {
  const { container } = render(
    <ToasterProvider>
      <Toaster />
    </ToasterProvider>
  );
  expect(container).toMatchSnapshot();
});
