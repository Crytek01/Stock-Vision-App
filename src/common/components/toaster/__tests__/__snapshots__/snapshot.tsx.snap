// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tab panel component 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft css-cpgvjg-MuiSnackbar-root"
    role="presentation"
  >
    <div
      direction="up"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
  </div>
</div>
`;
